core/src/worker/workflow/wft_poller.rs

```diff
@@ -121,26 +111,63 @@ impl WFTPollerShared {
             {
                 let mut sticky_active = sticky_active.clone();
                 let mut non_sticky_active = non_sticky_active.clone();
+                let mut sticky_backlog = self.last_seen_sticky_backlog.0.clone();
+
                 loop {
                     let num_sticky_active = *sticky_active.borrow_and_update();
                     let num_non_sticky_active = *non_sticky_active.borrow_and_update();
-                    let both_are_zero = num_sticky_active == 0 && num_non_sticky_active == 0;
-                    if both_are_zero {
+                    let num_sticky_backlog = *sticky_backlog.borrow_and_update();
+
+                    let allow = || {
```

Comment:
This could avoid being a closure by using else-ifs... but I don't have a firm position on if that's more readable or not.